tracer/src/Datadog.Trace/Configuration/ImmutableTracerSettings.cs

```diff
@@ -338,6 +339,11 @@ public ImmutableTracerSettings(TracerSettings settings)
         /// </remarks>
         internal int OutgoingTagPropagationHeaderMaxLength { get; }
 
+        /// <summary>
+        /// Gets a value indicating whether the rare sampler is enabled
+        /// </summary>
+        public bool IsRareSamplerEnabled { get; }
```

Comment:
We tend to make a lot of things internal ? Any good reason to make it public?

Reply:
If we make it internal then it can only be enabled through environment variables. This is a necessary limitation for some settings, but I feel like sampling should be configurable from the code

Reply:
We've been defaulting to making everything `internal` recently, as it dramatically simplifies the support story. For example, the rare sampler is already obsolete before we've added. If you make this public, we can't remove it _or even make it obsolete_ without it being a breaking change. And we shouldn't make breaking changes without a _very_ good reason. 

_And_ (thinking out loud here) customers _could_ configure it in code, by setting values in the configuration dictionary. Maybe we should be improving that experience rather than adding ad hoc properties for each feature? 🤔 


Also, this is in the `ImmutableTracerSettings`. Exposing it here doesn't make sampling configurable 😉

Reply:
Switching to internal then